scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -1614,12 +1628,12 @@ function Uninstall-Services([ref] $RestartNeeded, [bool] $LegacyInstaller) {
     if (Get-Service $EdgeServiceName -ErrorAction SilentlyContinue) {
         Set-Service -StartupType Disabled $EdgeServiceName -ErrorAction SilentlyContinue
         Stop-Service -NoWait -ErrorAction SilentlyContinue -ErrorVariable cmdErr $EdgeServiceName
-        if ($?) {
+        if ($? -and $LASTEXITCODE -eq 0) {
```

Comment:
`$LASTEXITCODE` is not set by PS commandlets. This will read an arbitrary stale value so it can fail spuriously.

```powershell
$ false; gci | Out-Null; echo $LASTEXITCODE

1
```

Reply:
right. for some reason I was thinking this script was using an internal function that invoked sc.exe. i'm still puzzled by why I've seen at least 3 times the stop-service succeed but the service was not actually stopped or even in 'stopping' state.

Reply:
One reason I know of is that if the service crashes, it gets restarted due to the failure recovery policy, and this takes precedence over any pending stop request. I've seen it happen on my VM a bunch of times. I work around it by doing a `Set-Service -StartupType Disabled` first.